plugins/parquet/parquetembed.cpp

```diff
@@ -1744,68 +1700,80 @@ void ParquetRecordBinder::processDecimal(const void *value, unsigned digits, uns
  */
 void ParquetRecordBinder::processUtf8(unsigned chars, const char *value, const RtlFieldInfo *field)
 {
-    bindUtf8Param(chars, value, field, r_parquet);
+    rapidjson::Value key = rapidjson::Value(field->name, jsonAlloc);
+    rapidjson::Value val = rapidjson::Value(value, rtlUtf8Size(chars, value), jsonAlloc);
+
+    parquetWriter->addMember(key, val);
 }
 
 /**
- * @brief Construct a new ParquetEmbedFunctionContext object
+ * @brief Construct a new ParquetEmbedFunctionContext object and parses the options set by the user.
  *
  * @param _logctx Context logger for use with the ParquetRecordBinder ParquetDatasetBinder classes.
+ * @param activityCtx Context about the Thor worker configuration.
  * @param options Pointer to the list of options that are passed into the Embed function.
  * @param _flags Should be zero if the embedded script is ok.
  */
 ParquetEmbedFunctionContext::ParquetEmbedFunctionContext(const IContextLogger &_logctx, const IThorActivityContext *activityCtx, const char *options, unsigned _flags)
-    : logctx(_logctx), m_scriptFlags(_flags)
+    : logctx(_logctx), scriptFlags(_flags)
 {
     // Option Variables
-    const char *option = "";      // Read(read), Read Parition(readpartition), Write(write), Write Partition(writepartition)
-    const char *location = "";    // file name and location of where to write parquet file
-    const char *destination = ""; // file name and location of where to read parquet file from
-    const char *partitionFields = ""; // comma delimited values containing fields to partition files on
-    __int64 rowsize = 40000;    // Size of the row groups when writing to parquet files
-    __int64 batchSize = 40000;  // Size of the batches when converting parquet columns to rows
-    bool overwrite = false;     // If true overwrite file with no error. The default is false and will throw an error if the file already exists.
-    arrow::Compression::type compressionOption = arrow::Compression::UNCOMPRESSED;
+    const char *option = "";            // Read(read), Read Parition(readpartition), Write(write), Write Partition(writepartition)
+    const char *location = "";          // file name and location of where to write parquet file
+    const char *destination = "";       // file name and location of where to read parquet file from
+    const char *partitionFields = "";   // Semicolon delimited values containing fields to partition files on
+    __int64 rowSize = 40000;            // Size of the row groups when writing to parquet files
+    __int64 batchSize = 40000;          // Size of the batches when converting parquet columns to rows
+    bool overwrite = false;             // If true overwrite file with no error. The default is false and will throw an error if the file already exists.
+    arrow::Compression::type compressionOption = arrow::Compression::UNCOMPRESSED; // Compression option that supports all arrow compression types.
```

Comment:
The comment is a bit confusing.  I think it is referring to 'what is compressionOption' but it somehow makes me think it is describing arrow::Compression::UNCOMPRESSED

Reply:
Reworded description of compressionOption.